Tasks/google-play-release/GooglePlay.js

```diff
@@ -227,29 +239,400 @@ function updateTrack(packageName, track, versionCode, userFraction) {
+                for (var i = 0; i < acceptedExtensions.length && !shouldAttemptUpload; i++) {
+                    var fullPathToFileToCheck = path.join(imageDirectory, imageType + acceptedExtensions[i]);
+                    try {
+                        var imageStat = fs.statSync(fullPathToFileToCheck);
+                        if (imageStat) {
+                            shouldAttemptUpload = imageStat.isFile();
+                            if (shouldAttemptUpload) {
+                                tl.debug(`Found image for type ${imageType} at ${fullPathToFileToCheck}`);
+                                imageList[imageType].push(fullPathToFileToCheck);
+                                break;
+                            }
+                        }
+                    } catch (e) {
+                        tl.debug(`File ${fullPathToFileToCheck} doesn't exist. Skipping...`);
+                    }
+                }
+
+                if (!shouldAttemptUpload) {
+                    tl.debug(`Image for ${imageType} was not found. Skipping...`);
+                }
+                break;
+            case "phoneScreenshots":
+            case "sevenInchScreenshots":
+            case "tenInchScreenshots":
+            case "tvScreenshots":
+            case "wearScreenshots":
+                try {
+                    var fullPathToDirToCheck = path.join(imageDirectory, imageType);
+                    var imageStat = fs.statSync(fullPathToDirToCheck);
+                    if (imageStat) {
+                        tl.debug(`Found something for type ${imageType}`);
+                        shouldAttemptUpload = imageStat.isDirectory();
+                        if (!shouldAttemptUpload) {
+                            tl.debug(`Stat returned that ${imageType} was not a directory. Is there a file that shares this name?`);
+                        } else {
+                            imageList[imageType] = fs.readdirSync(fullPathToDirToCheck).filter(function (image) {
+                                try {
+                                    pathIsDir = fs.statSync(path.join(fullPathToDirToCheck, image)).isFile();
+                                } catch (e) {
+                                    tl.debug(e);
+                                    tl.debug(`failed to stat path ${image}. ignoring...`);
+                                }
+
+                                return pathIsDir;
+                            });
+                        }
+                    }
+                } catch (e) {
+                    tl.debug(e);
+                    tl.debug(`Image directory for ${imageType} was not found. Skipping...`);
+                }
+                break;
+            default:
+                tl.debug(`Image type ${imageType} is an unknown type and was ignored`);
+                continue;
+        }
+    }
+
+    tl.debug(`Finished enumerating images: ${JSON.stringify(imageList)}`);
+    return imageList;
+}
```

Comment:
This looks good but perhaps `glob` could make your life easier to get the list of files under the directory for varied extensions, that could save time on looking up and stating individual file with different extension.

I will leave it up to you.